src/cmap/connection.ts

```diff
@@ -757,3 +769,534 @@ function write(
     operationDescription.cb();
   }
 }
+
+/** in-progress connection layer */
+
+/** @internal */
+export class ModernConnection extends TypedEventEmitter<ConnectionEvents> {
+  id: number | '<monitor>';
+  address: string;
+  socketTimeoutMS: number;
+  monitorCommands: boolean;
+  /** Indicates that the connection (including underlying TCP socket) has been closed. */
+  closed: boolean;
+  lastHelloMS?: number;
+  serverApi?: ServerApi;
+  helloOk?: boolean;
+  commandAsync: (
+    ns: MongoDBNamespace,
+    cmd: Document,
+    options: CommandOptions | undefined
+  ) => Promise<Document>;
+  /** @internal */
+  authContext?: AuthContext;
+
+  /**@internal */
+  [kDelayedTimeoutId]: NodeJS.Timeout | null;
+  /** @internal */
+  [kDescription]: StreamDescription;
+  /** @internal */
+  [kGeneration]: number;
+  /** @internal */
+  [kLastUseTime]: number;
+  /** @internal */
+  [kQueue]: Map<number, OperationDescription>;
+  /** @internal */
+  [kMessageStream]: MessageStream;
+  /** @internal */
+  socket: Stream;
+  /** @internal */
+  [kHello]: Document | null;
+  /** @internal */
+  [kClusterTime]: Document | null;
+
+  bufferPool: BufferPool;
```

Comment:
This doesn't exist on the Connection class so what other changes are there to this class?

Reply:
None, besides renaming `kStream` to `socket`